admin/securedrop_admin/__init__.py

```diff
@@ -599,11 +599,23 @@ def validate_gpg_keys(self) -> bool:
                 )
             except subprocess.CalledProcessError as e:
                 sdlog.debug(e.output)
-                raise FingerprintException(
-                    f"fingerprint {fingerprint} "
-                    + "does not match "
-                    + f"the public key {public_key}"
-                )
+                message = f"{fingerprint}: Fingerprint validation failed"
```

Comment:
The site-specific file has been created with the bad key and its fingerprint at this stage. It would be better if possible to fail early in sdconfig. and not create a valid config file with the unwanted key.

Reply:
Good point. This is an existing issue - I've filed https://github.com/freedomofpress/securedrop/issues/6950 to explain a few ways to fix it, and if my preferred way looks alright to you, I can tack on some commits to this PR to address both issues.

Reply:
> Good point. This is an existing issue - I've filed #6950 to explain a few ways to fix it, and if my preferred way looks alright to you, I can tack on some commits to this PR to address both issues.

I'm cool with treating it as a separate PR, tbh, it's a pre-existing bug. The validate-then-save approach you mention there seems good enough.